src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs

```diff
@@ -195,8 +197,14 @@ private void InitializeUsingRsa(RSA rsa, string algorithm)
             }
             else
             {
-                // default RSASignaturePadding for other supported RSA algorithms is Pkcs1
-                RSASignaturePadding = RSASignaturePadding.Pkcs1;
+                if (AppContext.TryGetSwitch(_enablePkcs1SignaturePaddingSwitch, out bool enablePaddingSwitch) && enablePaddingSwitch)
```

Comment:
I think it would be good to extract this into a separate helper method to simplify since it's always these two steps and since we always call this line when checking switches. Something like `bool SwitchEnabled(string switchName) => return bool`